### PR TITLE
Gstpe 36: Rest Module

### PR DESCRIPTION
**Pull request by isabelanog**
https://liferay.atlassian.net/browse/GSTPE-36

### CODE REVIEW

File: modules/amf-registration/amf-registration-rest/amf-registration-rest-impl/rest-openapi.yaml [hunk 1]
```diff
@@ -1,8 +1,199 @@
...
+                    type: string
+                state:
+                    description: The state.
+                    type: string
+                securityQuestion:
+                    description: Must choose one of the following
+                        What is your mother's maiden nae?
+                        What is the make of you first car?
+                        What is your high school mascot?
+                        Who is your favorite actor?
```
Comment (alexandresjunior) [inline modules/amf-registration/amf-registration-rest/amf-registration-rest-impl/rest-openapi.yaml]: Please, change description for something more succinct. the question might change in future, for instance.

Comment (isabelanog) [reply modules/amf-registration/amf-registration-rest/amf-registration-rest-impl/rest-openapi.yaml]: Done

File: modules/amf-registration/amf-registration-rest/amf-registration-rest-impl/src/main/java/com/liferay/amf/internal/resource/v1_0/AccountResourceImpl.java [hunk 2]
```diff
@@ -1,18 +1,41 @@
...
+
+	public Account createAccount(Account account) throws Exception {
+
+		return createAccount(account);
+	}
+
+	public Account updateAccount(Integer accountId, Account account) throws Exception {
+
+		return updateAccount(accountId,account);
+	}
```
Comment (alexandresjunior) [inline modules/amf-registration/amf-registration-rest/amf-registration-rest-impl/src/main/java/com/liferay/amf/internal/resource/v1_0/AccountResourceImpl.java]: you should call the `AccountLocalService` (the one you've created) and their CRUD methods respectively for each method defined in `AccountResourceImpl`. for that, you might need to include the `account-api` as a dependency in your `build.gradle` file. then, add something like this to call the service methods:

```
@Reference
private AccountLocalService _accountLocalService;
```

let me know if you have any other questions ;)

File: modules/amf-registration/amf-registration-service/service.xml [hunk 3]
```diff
@@ -8,7 +8,7 @@
 
 		<!-- PK fields --> 
 
-		<column name="accountId" primary="true" type="long" />
+		<column name="accountId" primary="true" type="String" />
```
Comment (alexandresjunior) [inline modules/amf-registration/amf-registration-service/service.xml:11]: it is a best practice to use `long` for Ids

File: modules/amf-registration/amf-registration-rest/amf-registration-rest-impl/src/main/java/com/liferay/amf/internal/resource/v1_0/AccountResourceImpl.java [hunk 4]
```diff
@@ -21,33 +17,34 @@
...
 
 	@Override
-	public Account getAccount(String accountId, Account account) throws Exception {
+	public Account getAccount(String accountId) throws Exception {
 
-		return _accountLocalService.getAccount(accountId, account.getFirstName(), account.getLastName(), account.getEmailAdress(),
-				account.getUserName(), account.getGenre(), account.getBirthday(), account.getBirthday(), account.getPassword(), account.getHomePhone(),
-				account.getMobilePhone(), account.getAdress1(),account.getAdress2(), account.getCity(), account.getState(), account.getSecurityQuestion(),
-				account.getSecurityAnswer(), account.getacceptedTou);
+		return (Account) _accountLocalService.getAccount(accountId);
```
Comment (alexandresjunior) [inline modules/amf-registration/amf-registration-rest/amf-registration-rest-impl/src/main/java/com/liferay/amf/internal/resource/v1_0/AccountResourceImpl.java]: casting types might lead to errors. I would recommend to create a `mapper` class and include within it a `toDTO` method to convert `Account` (model) to `Account` (DTO).

File: modules/amf-registration/amf-registration-rest/amf-registration-rest-impl/src/main/java/com/liferay/amf/internal/resource/v1_0/AccountResourceImpl.java [hunk 5]
```diff
@@ -21,33 +17,34 @@
...
 
-	@Override
+
 	public Account createAccount(Account account) throws Exception {
 
-		return _accountLocalService.createAccount(account.getFirstName(), account.getLastName(), account.getEmailAdress(),
-				account.getUserName(), account.getGenre(), account.getBirthday(), account.getBirthday(), account.getPassword(), account.getHomePhone(),
-				account.getMobilePhone(), account.getAdress1(),account.getAdress2(), account.getCity(), account.getState(), account.getSecurityQuestion(),
-				account.getSecurityAnswer(), account.getacceptedTou);
+		return (Account) _accountLocalService.createAccount(account.getFirstName(), account.getLastName(),
```
Comment (alexandresjunior) [inline modules/amf-registration/amf-registration-rest/amf-registration-rest-impl/src/main/java/com/liferay/amf/internal/resource/v1_0/AccountResourceImpl.java]: same as above.

File: modules/amf-registration/amf-registration-rest/amf-registration-rest-impl/src/main/java/com/liferay/amf/internal/resource/v1_0/AccountResourceImpl.java [hunk 6]
```diff
@@ -21,33 +17,34 @@
...
-	@Override
+
+
 	public Account updateAccount(String accountId, Account account) throws Exception {
 
-		return _accountLocalService.updateAccount(accountId, account.getFirstName(), account.getLastName(), account.getEmailAdress(),
-				account.getUserName(), account.getGenre(), account.getBirthday(), account.getBirthday(), account.getPassword(), account.getHomePhone(),
-				account.getMobilePhone(), account.getAdress1(),account.getAdress2(), account.getCity(), account.getState(), account.getSecurityQuestion(),
-				account.getSecurityAnswer(), account.getacceptedTou);
+		return (Account) _accountLocalService.updateAccount(accountId, account.getFirstName(), account.getLastName(),
```
Comment (alexandresjunior) [inline modules/amf-registration/amf-registration-rest/amf-registration-rest-impl/src/main/java/com/liferay/amf/internal/resource/v1_0/AccountResourceImpl.java]: same as above.

File: modules/amf-registration/amf-registration-rest/amf-registration-rest-impl/src/main/java/com/liferay/amf/rest/internal/resource/v1_0/AccountResourceImpl.java [hunk 7]
```diff
@@ -13,4 +16,32 @@
...
+			account.getMobilePhone(),
+			account.getAddress1(),
+			account.getAddress2(),
+			account.getCity(),
+			account.getState(),
+			account.getSecurityQuestion(),
+			account.getSecurityAnswer()
+		);
+
+		return account;
```
Comment (alexandresjunior) [inline modules/amf-registration/amf-registration-rest/amf-registration-rest-impl/src/main/java/com/liferay/amf/rest/internal/resource/v1_0/AccountResourceImpl.java]: use the `mapper` class to convert the `Account` (model) to `Account` (DTO) and return this last one. so we make sure everything was assigned and returned properly (my opinion).

**Review by alexandresjunior — CHANGES_REQUESTED** [review]
Added a few more comments. Let me know if you have any more questions, thanks!  ;)

**Comment by isabelanog** [comment]
GSTPE-36: Rest Module Review

**Pull request merged**